### ISSUE TITLE
Redirecting back to own profile when tapping friend's profile in the search box

### ISSUE THREAD

**Issue by dbui04** [AllanWang/Frost-for-Facebook#1688]
- [x] I have verified that the bug does not occur in my mobile browser (at https://touch.facebook.com/)

**Describe the bug**

**To Reproduce**
Steps to reproduce the behaviour:

**Details (please provide at least the app version):**
 - App Version: 2.4.5
 - Device: Nokia 8.1
 - Android Version: 10

**Comment by AllanWang** [comment]
Cannot replicate.

**Comment by GibberMeJenks** [comment]
@AllanWang I get this problem occasionally. For some reason it redirects to your own profile for some people you search while for others it goes to the correct profile you are looking for. Very odd.

**Comment by AllanWang** [comment]
Is there any pattern you saw? Are the differences happening with different queries or with different entries in a single query?

**Comment by GibberMeJenks** [comment]
It's only happened about twice - one time when I was searching for a friend, and the other when I searched for a friend to add.

I tried testing this just now and searched both friends and random people and clicked on profiles to recreate, but it works fine. However, I did find one profile (herein John Doe) that consistently has this error, everyone else i searched had no issues.

John Doe shares the same first letter in their first and last name as me. I then tried searching a random name with the same first/last letter as my first/last name but there was also no issue oddly enough - it was only their specific profile where it redirects back to mine after searching their name and clicking their profile. 

Also did a search for the exact name as John Doe but a different individual - again, no error.
